From 33f39f780507c1df1a0d815fbb9e5a8a432a3a30 Mon Sep 17 00:00:00 2001
From: Xilin Wu <sophon@radxa.com>
Date: Wed, 22 Apr 2026 23:35:58 +0800
Subject: [PATCH 075/268] dt-bindings: net: qca808x: document qca,led-mode

Add a qca,led-mode DT property for QCA8081 PHYs to select one of
the datasheet-defined LED preset modes.

The property programs the MMD7 LED control registers for the three
PHY LED outputs during PHY initialization and accepts values 1..3.

Update the binding example to show a PHY using LED mode 2.

Signed-off-by: Xilin Wu <sophon@radxa.com>
---
 .../devicetree/bindings/net/qca,qca808x.yaml         | 12 ++++++++++++
 1 file changed, 12 insertions(+)

--- a/Documentation/devicetree/bindings/net/qca,qca808x.yaml
+++ b/Documentation/devicetree/bindings/net/qca,qca808x.yaml
@@ -25,6 +25,17 @@ properties:
     enum:
       - ethernet-phy-id004d.d101
 
+  qca,led-mode:
+    $ref: /schemas/types.yaml#/definitions/uint32
+    description: |
+      Select one of the vendor-defined LED preset modes described in the
+      datasheet. The selected mode programs the MMD7 LED control registers for
+      all three PHY LED outputs during PHY initialization.
+    enum:
+      - 1
+      - 2
+      - 3
+
 unevaluatedProperties: false
 
 examples:
@@ -38,6 +49,7 @@ examples:
         ethernet-phy@0 {
             compatible = "ethernet-phy-id004d.d101";
             reg = <0>;
+            qca,led-mode = <2>;
 
             leds {
                 #address-cells = <1>;
